File: app/src/main/java/net/oschina/git/roland/wimm/main/WelcomeActivity.java
package net.oschina.git.roland.wimm.main;

import android.content.Intent;
import android.os.Handler;
import android.os.Message;

import net.oschina.git.roland.wimm.R;
import net.oschina.git.roland.wimm.common.base.BaseActivity;
import net.oschina.git.roland.wimm.login.LoginActivity;

import org.xutils.view.annotation.ContentView;

/**
 * Created by Roland on 2017/4/10.
 */
@ContentView(R.layout.activity_welcome)
public class WelcomeActivity extends BaseActivity {

    private Handler welcomeHandler = new Handler(new Handler.Callback() {
        @Override
        public boolean handleMessage(Message msg) {
            Intent intent = new Intent(WelcomeActivity.this, LoginActivity.class);
            startActivity(intent);
            finish();
            return true;
        }
    });

    @Override
    protected void initComp() {
        welcomeHandler.sendEmptyMessageDelayed(0, 1000);
    }

    @Override
    protected void initListener() {

    }

    @Override
    protected void initData() {

    }
}

File: app/src/main/java/net/oschina/git/roland/wimm/main/WelcomeActivity.kt
package net.oschina.git.roland.wimm.main

import android.content.Intent
import android.os.Bundle
import android.os.Handler

import net.oschina.git.roland.wimm.R
import net.oschina.git.roland.wimm.common.base.BaseActivity

/**
 * Created by Roland on 2017/4/10.
 */
class WelcomeActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        kotlin = true
        super.onCreate(savedInstanceState)
    }

    private val welcomeHandler = Handler(Handler.Callback {
        val intent = Intent(this@WelcomeActivity, LoginActivity::class.java)
        startActivity(intent)
        finish()
        true
    })

    override fun getContentViewLayout(): Int {
        return R.layout.activity_welcome
    }

    override fun initComp() {
        welcomeHandler.sendEmptyMessageDelayed(0, 1000)
    }

    override fun initListener() {

    }

    override fun initData() {

    }
}
